lib/core/logger.cpp

```diff
@@ -0,0 +1,57 @@
+#include <spdlog/common.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/spdlog.h>
+#include <tl/expected.hpp>
+
+
+namespace {
+
+tl::expected<spdlog::level::level_enum, std::string> parseLoggerLevel(std::string levelStr) {
+    std::transform(levelStr.begin(), levelStr.end(), levelStr.begin(), [](unsigned char c) {
+        return std::tolower(c);
+    });
+
+    // TODO: magic_enum can hide these ifs
+    if (levelStr == "trace") {
+        return spdlog::level::trace;
+    }
+
+    if (levelStr == "debug") {
+        return spdlog::level::debug;
+    }
+
+    if (levelStr == "info") {
+        return spdlog::level::info;
+    }
+
+    if (levelStr == "warn") {
+        return spdlog::level::warn;
+    }
+
+    if (levelStr == "err") {
+        return spdlog::level::err;
+    }
+
+    if (levelStr == "critical") {
+        return spdlog::level::critical;
+    }
+    return tl::make_unexpected("Failed to parse logger level name");
+}
+
+__attribute__((constructor)) void initLogger() {
+    auto log_level_env = std::getenv("OKL_LOGGER_LEVEL");
```

Comment:
Maybe `OKL_VERBOSE` ?

Reply:
{TOOL-NAME}_LOG_LEVEL is quite cannonical way for naming such env var

Reply:
Why? It has multiple level, verbose is boolean by nature?

Reply:
> {TOOL-NAME}_LOG_LEVEL is quite cannonical way for naming such env var I propose OKLT_LOG_LEVEL

Ok